controllers/precache.go

```diff
@@ -344,9 +344,13 @@ func (r *ClusterGroupUpgradeReconciler) extractPrecachingSpecFromPolicies(
 			kind := object["kind"]
 			switch kind {
 			case utils.PolicyTypeClusterVersion:
-				image := object["spec"].(map[string]interface {
-				})["desiredUpdate"].(map[string]interface{})["image"]
-				if image != nil {
+				cvSpec := object["spec"].(map[string]interface{})
+				desiredUpdate, found := cvSpec["desiredUpdate"]
+				if !found {
+					continue
+				}
+				image, found := desiredUpdate.(map[string]interface{})["image"]
+				if found && image != "" {
```

Comment:
We need to do value check on these parameters as well. 
				} else {
					upstream := object["spec"].(map[string]interface {
					})["upstream"].(string)
					channel := object["spec"].(map[string]interface {
					})["channel"].(string)
					version := object["spec"].(map[string]interface {
					})["desiredUpdate"].(map[string]interface{})["version"].(string)

Reply:
Yes, they are checked inside function `getImageForVersionFromUpdateGraph`. If one of them is omitted, PrecacheSpecValidCondition is set to false with reason PlatformImageInvalid